src/Microsoft.ML/LearningPipeline.cs

```diff
@@ -48,15 +49,26 @@ public ScorerPipelineStep(Var<IDataView> data, Var<ITransformModel> model)
     [DebuggerTypeProxy(typeof(LearningPipelineDebugProxy))]
     public class LearningPipeline : ICollection<ILearningPipelineItem>
     {
+        readonly internal IHostEnvironment Env;
         private List<ILearningPipelineItem> Items { get; } = new List<ILearningPipelineItem>();
 
         /// <summary>
         /// Construct an empty <see cref="LearningPipeline"/> object.
         /// </summary>
-        public LearningPipeline()
+        public LearningPipeline(int? seed = null, int concurrency = 0)
         {
+
+            var env = new DefaultEnvironment(seed: seed, conc: concurrency);
+            env.MessageRecieved += Env_MessageRecieved;
+            Env = env;
+        }
+
+        private void Env_MessageRecieved(object sender, ChannelMessageEventArgs e)
+        {
+            MessageOccured?.Invoke(this, e);
         }
 
+        public event EventHandler<ChannelMessageEventArgs> MessageOccured;
```

Comment:
This is a good example of how this class is morphing to become host environment like. You create a facility identical to that of the environment you create, but you choose to rename it from `MessageReceived` to `MessageOccured`.